R/plot_section.R

```diff
@@ -0,0 +1,154 @@
+#' tax_range_strat(occdf, certainty = "certainty", by = "LAD")
+#' tax_range_strat(occdf, certainty = "certainty", by = "name")
+#'
+#' @export
+tax_range_strat <- function (occdf, name = "taxon", level = "bed",
+                             certainty = FALSE, by = "FAD",
+                             units = "height", label = "Section A")
+{
+  if (is.data.frame(occdf) == FALSE) {
+    stop("`occdf` should be a dataframe")
+  }
+
+  if (!is.numeric(occdf[, level])) {
+    stop("`level` must be of class numeric.")
+  }
+
+  if (any(c(name, level) %in% colnames(occdf) == FALSE)) {
+    stop("Either `name` or `level` is not a named column in\n`occdf`")
+  }
+
+  if (certainty != FALSE && certainty %in% colnames(occdf) == FALSE) {
+    stop("`certainty` is not a named column in\n`occdf`")
+  }
+
+  if (any(is.na(occdf[, name]))) {
+    stop("The `name` column contains NA values")
+  }
+
+  if (any(is.na(occdf[, level]))) {
+    stop("The `level` column contains NA values")
+  }
+
+  if (!by %in% c("name", "FAD", "LAD")) {
+    stop("`by` must be either \"FAD\", \"LAD\", or \"name\"")
+  }
+
+  if (!units %in% c("height", "beds")) {
+    stop("`units` must be either \"height\" or \"beds\"")
+  }
+
+  #List and order unique taxa
+  unique_taxa <- unique(occdf[, name])
+  unique_taxa <- unique_taxa[order(unique_taxa)]
+
+  #Create object to hold information
+  ranges <- data.frame(taxon = unique_taxa, min_bin = NA, max_bin = NA)
+
+  #Populate nested list
+  for (i in seq_along(unique_taxa)) {
+    occ_filter <- occdf[(occdf[, name] == unique_taxa[i]),]
+    ranges[i,2] <- min(occ_filter[level])
+    ranges[i,3] <- max(occ_filter[level])
+  }
+
+  #Reorder lists
+  if (by == "LAD") {
+    ranges <- ranges[order(ranges$min_bin), ]
+    ranges <- ranges[order(ranges$max_bin), ]
+  }
+
+  if (by == "FAD") {
+    ranges <- ranges[order(ranges$max_bin), ]
+    ranges <- ranges[order(ranges$min_bin), ]
+  }
+
+  #Add ID numbers
+  ranges$ID <- c(1:length(unique_taxa))
+  labels <- ranges[, c(1,4)]
+  occdf <- merge(occdf, labels, by.x = name, by.y = "taxon")
+
+  #Split certain and uncertain occurrences
+  if (certainty == FALSE){
+  certain <- occdf
+  } else {
+  certain <- occdf[(occdf[, certainty] == 1), ]
+  uncertain <- occdf[(occdf[, certainty] == 0), ]
+  }
+
+  #Determine y-axis label
+  if (units == "height") {
+    y_axis <- "Stratigraphic height (m)"
+  } else
+    y_axis <- "Bed number"
+
+  #Create plot
+  plot(x = NA, y = NA,
+       xlim = c(1, length(unique_taxa)),
+       ylim = c(min(ranges$min_bin), max(ranges$max_bin)),
+       axes = FALSE,
+       xlab = " ",
+       ylab = y_axis,
+       main = label)
+  segments(y0 = ranges$min_bin, y1 = ranges$max_bin,
+           x0 = ranges$ID,
+           col = "black")
+  points(y = certain$bed, x = certain$ID, pch = 19,
+         col = "black")
+  if (certainty != FALSE){
+    points(y = uncertain$bed, x = uncertain$ID, pch = 1,
+          col = "black")
+  }
```

Comment:
This might be personal preference, but I really don't like seeing the lines peak out from behind the circles. I'd recommend having filled circles (with white fills). But again, how much of this should be left to the user to define?